lib/index.js

```diff
@@ -111,6 +111,57 @@ class LilyWeight {
         } else return { uuid, ...weight };
     }
 
+    /**
+     * Returns the specified profile's Weight.
+     * @param {string} uuid - Player's Minecraft UUID, can be with dashes.
+     * @param  {string} specifiedProfile - The profile name, can be uppercase or lowercase.
+     * @param  {boolean} [returnUsername=false] - Should the function return the username.
+     * @returns {Promise<PlayerWeights>} The weights calculated for the player.
+     */
+     async getProfileWeightFromUUID (uuid, specifiedProfile, returnUsername = false) {
+        if (uuid.length === 36)
+            uuid = uuid.replace(/-/g, "");
+
+        let username;
+        const profile = await getSpecifiedProfile(uuid, this.#apiKey, specifiedProfile);
+
+        if (!profile) throw Error("Profile name invalid");
+
+        const slayerXP = [
+            profile.slayer_bosses?.zombie?.xp ?? 0,
+            profile.slayer_bosses?.spider?.xp ?? 0,
+            profile.slayer_bosses?.wolf?.xp ?? 0,
+            profile.slayer_bosses?.enderman?.xp ?? 0
+        ];
+
+        const cataCompl = profile.dungeons?.dungeon_types?.catacombs?.tier_completions ?? {};
+        const mCataCompl = profile.dungeons?.dungeon_types?.master_catacombs?.tier_completions ?? {};
+        const cataXP = profile.dungeons?.dungeon_types?.catacombs?.experience ?? 0;
+
+        let skillLevels;
+        let skillXP;
+        if (!profile.experience_skill_mining) {
+            // skill API is off
+            const player = await getPlayer(uuid, this.apiKey);
```

Comment:
I've just noticed this but this should be `this.#apiKey`, you're missing the `#`.